File: app/src/main/java/org/d1scw0rld/bookbag/dto/Field.java
package org.d1scw0rld.bookbag.dto;

/**
 * Created by Iasen on 12.7.2016 г..
 */

public class Field
{
   public  int     id;
   public  String  name;
   public  boolean isVisible  = false;
   public  byte    type;
   public  int     inputType = 0;

   public final static byte TYPE_TEXT              = 1,
                            TYPE_TEXT_AUTOCOMPLETE = 2,
                            TYPE_MONEY             = 3,
                            TYPE_MULTIFIELD        = 4,
                            TYPE_SPINNER           = 5,
                            TYPE_MULTI_SPINNER     = 6,
                            TYPE_DATE              = 7,
                            TYPE_RATING            = 8,
                            TYPE_CHECK_BOX         = 9;
   
   
   public Field(int id, String name, boolean isVisible, byte type)
   {
      this.id = id;
      this.name = name;
      this.isVisible = isVisible;
      this.type = type;
   }

   public Field(int id, String name, byte type)
   {
      this.id = id;
      this.name = name;      
      this.type = type;
   }

   public Field setVisibility(boolean isVisible)
   {
      this.isVisible = isVisible;
      return this;
   }
   
   public Field setInputType(int inputType)
   {
      this.inputType = inputType;
      return this;
   }
}

File: app/src/main/java/org/d1scw0rld/bookbag/dto/Field.kt
package org.d1scw0rld.bookbag.dto

class Field @JvmOverloads constructor(
    @JvmField var id: Int = 0,
    @JvmField var name: String = "",
    @JvmField var isVisible: Boolean = false,
    @JvmField var type: Byte = 0
) {
    @JvmField var inputType: Int = 0

    companion object {
        const val TYPE_TEXT: Byte = 1
        const val TYPE_TEXT_AUTOCOMPLETE: Byte = 2
        const val TYPE_MONEY: Byte = 3
        const val TYPE_MULTIFIELD: Byte = 4
        const val TYPE_SPINNER: Byte = 5
        const val TYPE_MULTI_SPINNER: Byte = 6
        const val TYPE_DATE: Byte = 7
        const val TYPE_RATING: Byte = 8
        const val TYPE_CHECK_BOX: Byte = 9
    }

    constructor(id: Int, name: String, type: Byte) : this(id, name, false, type)

    fun setVisibility(isVisible: Boolean): Field {
        this.isVisible = isVisible
        return this
    }

    fun setInputType(inputType: Int): Field {
        this.inputType = inputType
        return this
    }
}
